llvm/lib/Transforms/IPO/MemProfContextDisambiguation.cpp

```diff
@@ -1694,11 +1682,10 @@ void CallsiteContextGraph<DerivedCCG, FuncTy,
   MapVector<CallInfo, ContextNode *> TailCallToContextNodeMap;
 
   for (auto Entry = NonAllocationCallToContextNodeMap.begin();
-       Entry != NonAllocationCallToContextNodeMap.end();) {
+       Entry != NonAllocationCallToContextNodeMap.end(); Entry++) {
```

Comment:
While you are here, may I suggest a range-based for loop here?

Reply:
Yeah, good idea, we can do that now. My local branch seems to have gotten messed up, so I am going to merge this one, then make a separate NFC follow on change to do that.

Reply:
Done in https://github.com/llvm/llvm-project/pull/94308